third_party/terraform/resources/resource_storage_bucket.go

```diff
@@ -873,8 +959,38 @@ func resourceGCSBucketLifecycleRuleConditionHash(v interface{}) int {
 		buf.WriteString(fmt.Sprintf("%s-", v.(string)))
 	}
 
-	if v, ok := m["is_live"]; ok {
-		buf.WriteString(fmt.Sprintf("%t-", v.(bool)))
+	// Note that we are keeping the boolean notation from when is_live was
+	// the only field (i.e. not deprecated) in order to prevent a diff from
+	// hash key.
+	// There are three possible states for the actual condition
+	// and correspond to the following hash codes:
+	//
+	// 1. LIVE only: "true-"
+	//    Applies for one of:
+	//      with_state = "" && is_live = true
+	//      with_state = "LIVE"
+	//
+	// 2. ARCHIVED only: "false-"
+	//    Applies for one of:
+	//      with_state = "" && is_live = false
+	//      with_state = "ARCHIVED"
+	//
+	// 3. ANY (i.e. LIVE and ARCHIVED): ""
+	//    Applies for one of:
+	//      with_state = "ANY"
+
+	withStateV, withStateOk := m["with_state"]
+	if !withStateOk || withStateV == "" {
```

Comment:
I think you can simplify to `withStateV == ""` since if it's an empty string it doesn't matter if it was present or not. Then below can just be `else`

Reply:
I'd rather keep the withStateOk since it's a map[string]interface{} - I do actually need to cast it though, so I'll change that.